### PR TITLE
Refactor preparing `package.json` files

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
I found a fair bit of duplicated logic when updating NPM and Yarn lockfiles that's already implemented in `PackageJsonPreparer`. So I'm refactoring things to reuse it.

### CODE REVIEW

File: npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb [hunk 1]
```diff
@@ -574,18 +537,6 @@ def yarnrc_content
...
 
-        def sanitized_package_json_content(content)
-          updated_content =
-            content.
-            gsub(/\{\{[^\}]*?\}\}/, "something"). # {{ nm }} syntax not allowed
-            gsub(/(?<!\\)\\ /, " ").          # escaped whitespace not allowed
-            gsub(%r{^\s*//.*}, " ")           # comments are not allowed
-
-          json = JSON.parse(updated_content)
-          json["name"] = json["name"].delete(" ") if json["name"].is_a?(String)
```
Comment (deivid-rodriguez) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb:585]: This cleanup is no longer done after this refactoring. I think it's no longer needed but I'll extract its removal to a separate PR.

Comment (deivid-rodriguez) [reply npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb]: This was extracted into https://github.com/dependabot/dependabot-core/pull/7246 and deployed a couple of weeks ago.

File: npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb [hunk 2]
```diff
@@ -545,6 +492,22 @@ def updated_package_json_content(file)
             ).updated_package_json.content
         end
 
+        def prepared_package_json_content(content)
+          @prepared_package_json_content ||= {}
+          @prepared_package_json_content[content] ||=
+            package_json_preparer(
+              content
+            ).prepared_content
+        end
```
Comment (Nishnha) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb:501]: Adding this as a memoized function seems extra compared to using `package_json_preparer(content).prepared_content` explicitly?

I don't feel strongly about this though so it is also fine to leave in.

**Review by Nishnha — APPROVED** [review]
Thanks!

File: npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb (hunk 2 above)
Comment (deivid-rodriguez) [reply npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb]: I removed the extra memoization and also found a similar one which I removed too at https://github.com/dependabot/dependabot-core/pull/7245/commits/81da5a1b40699053079e26730822daa94845160c.

**Pull request merged**